libraries/apollo-runtime-java/src/main/java/com/apollographql/apollo3/runtime/java/interceptor/ApolloDisposable.java

```diff
@@ -4,4 +4,12 @@ public interface ApolloDisposable {
   void dispose();
 
   boolean isDisposed();
+
+  void addListener(Listener listener);
+
+  void removeCancellationListener(Listener listener);
+
+  interface Listener {
+    void onCancelled();
```

Comment:
Maybe `onDisposed` (or `onDispose`) would make more sense?

Reply:
100%. I wanted to do this actually and forgot to finish it 🤦